python/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK/ALB_HTTP_TO_HTTPS_REDIRECTION_CHECK_test.py

```diff
@@ -0,0 +1,199 @@
+            return_value={'LoadBalancers': [{'LoadBalancerArn': 'arn1', 'Type': 'application'}]}
+        )
+        ELBV2_CLIENT_MOCK.describe_listeners = MagicMock(
+            return_value={'Listeners': [{'ListenerArn': 'arn1', 'SslPolicy': 'Some_policy_1'}, {'ListenerArn': 'arn2', 'SslPolicy': 'Some_policy_2'}]}
+        )
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        assert_successful_evaluation(self, response, [build_expected_response('COMPLIANT', 'arn1')], 1)
+
+    def test_scenario_4_compliant(self):
+        ELBV2_CLIENT_MOCK.describe_load_balancers = MagicMock(
+            return_value={'LoadBalancers': [{'LoadBalancerArn': 'arn1', 'Type': 'application'}, {'LoadBalancerArn': 'arn2', 'Type': 'application'}]}
+        )
+        ELBV2_CLIENT_MOCK.describe_listeners = MagicMock(
+            return_value={'Listeners': [
+                {'ListenerArn': 'arn1', 'DefaultActions': [{'RedirectConfig': {'Protocol': 'HTTPS'}, 'Type': 'redirect'}]},
+                {'ListenerArn': 'arn2', 'DefaultActions': [{'RedirectConfig': {'Protocol': 'HTTPS'}, 'Type': 'redirect'}]},
+            ]}
+        )
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        resp_expected = [build_expected_response('COMPLIANT', 'arn1'), build_expected_response('COMPLIANT', 'arn2')]
+        assert_successful_evaluation(self, response, resp_expected, 2)
+
+class NonCompliantResourceTest(unittest.TestCase):
+    def test_scenario_3_non_compliant(self):
+        ELBV2_CLIENT_MOCK.describe_load_balancers = MagicMock(
+            return_value={'LoadBalancers': [{'LoadBalancerArn': 'arn1', 'Type': 'application'}, {'LoadBalancerArn': 'arn2', 'Type': 'application'}]}
+        )
+        ELBV2_CLIENT_MOCK.describe_listeners = MagicMock(
+            return_value={'Listeners': [
+                {'ListenerArn': 'arn1', 'DefaultActions': [{'RedirectConfig': {'Protocol': 'HTTP'}, 'Type': 'other'}]},
+                {'ListenerArn': 'arn2', 'DefaultActions': [{'RedirectConfig': {'Protocol': 'HTTP'}, 'Type': 'other'}]},
+            ]}
+        )
+        response = RULE.lambda_handler(build_lambda_scheduled_event(), {})
+        resp_expected = []
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'arn1', annotation="HTTP listener rule must have HTTP to HTTPS redirection action configured"))
+        resp_expected.append(build_expected_response('NON_COMPLIANT', 'arn2', annotation="HTTP listener rule must have HTTP to HTTPS redirection action configured"))
+        assert_successful_evaluation(self, response, resp_expected, 2)
+
+class NotApplicableResourceTest(unittest.TestCase):
+    def test_scenario_1_not_applicable(self):
+        ELBV2_CLIENT_MOCK.describe_load_balancers = MagicMock(
+            return_value={'LoadBalancers': [{'LoadBalancerArn': 'arn1', 'Type': 'other'}]}
+        )
+        ELBV2_CLIENT_MOCK.describe_listeners = MagicMock(
```

Comment:
`describe_listeners` need not be mocked as the code goes past this API call if there is no ALB